dojo/tools/acunetix/parser.py

```diff
@@ -21,38 +22,30 @@ def __init__(self, xml_output, test):
         self.set_defectdojo_findings(acunetix_defectdojo_findings, test)
 
     def set_defectdojo_findings(self, acunetix_defectdojo_findings, test):
-        defectdojo_findings = []
+        defectdojo_findings = dict()
 
         for acunetix_defectdojo_finding in acunetix_defectdojo_findings:
+            dupe_key = hashlib.md5((acunetix_defectdojo_finding.title + acunetix_defectdojo_finding.description).encode("utf-8")).hexdigest()
 
-            defectdojo_title = acunetix_defectdojo_finding.title
-            defectdojo_date = get_defectdojo_date(acunetix_defectdojo_finding.date)
-            defectdojo_cwe_number = get_cwe_number(acunetix_defectdojo_finding.cwe)
-            defectdojo_severity = get_severity(acunetix_defectdojo_finding.severity)
-            defectdojo_falsep = get_false_positive(acunetix_defectdojo_finding.false_p)
-
-            defectdojo_findings_titles = [finding.title for finding in defectdojo_findings]
-
-            if defectdojo_title not in defectdojo_findings_titles:
-                finding = Finding(
-                            title=defectdojo_title,
-                            date=defectdojo_date,
+            if dupe_key not in defectdojo_findings:
+                defectdojo_findings[dupe_key] = Finding(
+                            title=acunetix_defectdojo_finding.title,
+                            date=get_defectdojo_date(acunetix_defectdojo_finding.date),
                             url=acunetix_defectdojo_finding.url,
-                            cwe=defectdojo_cwe_number,
+                            cwe=get_cwe_number(acunetix_defectdojo_finding.cwe),
                             test=test,
-                            severity=defectdojo_severity,
+                            severity=get_severity(acunetix_defectdojo_finding.severity),
                             description=acunetix_defectdojo_finding.description,
                             mitigation=acunetix_defectdojo_finding.mitigation,
                             references=acunetix_defectdojo_finding.references,
                             impact=acunetix_defectdojo_finding.impact,
-                            false_p=defectdojo_falsep,
+                            false_p=get_false_positive(acunetix_defectdojo_finding.false_p),
                             dynamic_finding=acunetix_defectdojo_finding.dynamic_finding
                 )
-                defectdojo_findings.append(finding)
             else:
-                print(("Duplicate finding : {defectdojo_title}".format(defectdojo_title=defectdojo_title)))
+                print(("Duplicate finding : {defectdojo_title}".format(defectdojo_title=acunetix_defectdojo_finding.title)))
```

Comment:
can this be turned into a debug log statement, or maybe even removed?

Reply:
I can change that